### PR TITLE
Fortinet FortiManager HA Configuration Module

### PR DESCRIPTION
**Pull request by ftntcorecse**
##### SUMMARY

This module allows the configuration of high-availability (HA) for FortiManager nodes.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

fmgr_ha.py
test_fmgr_ha.py
fixtures/test_fmgr_ha.py
##### ANSIBLE VERSION

```paste below
ansible 2.4.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/usr/ansible_modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/network/fortimanager/fmgr_ha.py [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+    - Luke Weighall (@lweighall)
+    - Andrew Welsh (@Ghilli3)
+    - Jim Huber (@p4r4n0y1ng)
+short_description: Manages the High-Availability State of FortiManager Clusters and Nodes
+description: Change HA state or settings of FortiManager nodes (Standalone/Master/Slave)
+
+options:
+  host:
+    description:
+      - The FortiManager's Address.
```
Comment (samccann) [inline lib/ansible/modules/network/fortimanager/fmgr_ha.py]:
```suggestion
      - The FortiManager's address.
```

File: lib/ansible/modules/network/fortimanager/fmgr_ha.py [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+    required: true
+  username:
+    description:
+      - The username to log into the FortiManager
```
Comment (samccann) [inline lib/ansible/modules/network/fortimanager/fmgr_ha.py]:
```suggestion
      - The username to log into the FortiManager.
```

File: lib/ansible/modules/network/fortimanager/fmgr_ha.py [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+    required: true
+  password:
+    description:
+      - The password associated with the username account.
+    required: false
+  fmgr_ha_mode:
+    description:
+      - Sets the role of the FortiManager host for HA
```
Comment (samccann) [inline lib/ansible/modules/network/fortimanager/fmgr_ha.py]:
```suggestion
      - Sets the role of the FortiManager host for HA.
```

File: lib/ansible/modules/network/fortimanager/fmgr_ha.py [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+    description:
+      - Sets the IPv4 address of a HA peer.
+    required: false
+  fmgr_ha_peer_ipv6:
+    description:
+      - Sets the IPv6 address of a HA peer.
+    required: false
+  fmgr_ha_peer_sn:
+    description:
+      - Sets the HA Peer Serial Number
```
Comment (samccann) [inline lib/ansible/modules/network/fortimanager/fmgr_ha.py]:
```suggestion
      - Sets the HA Peer Serial Number.
```

File: lib/ansible/modules/network/fortimanager/fmgr_ha.py [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+    required: false
+  fmgr_ha_peer_status:
+    description:
+      - Sets the peer status enable or disable
```
Comment (samccann) [inline lib/ansible/modules/network/fortimanager/fmgr_ha.py]:
```suggestion
      - Sets the peer status to enable or disable.
```

File: lib/ansible/modules/network/fortimanager/fmgr_ha.py [hunk 6]
```diff
@@ -0,0 +1,404 @@
...
+    required: false
+    choices: ["enable", "disable"]
+  fmgr_ha_cluster_pw:
+    description:
+      - Sets the password for the HA cluster. Only required once. System remembers between HA mode switches.
+    required: false
+  fmgr_ha_cluster_id:
+    description:
+      - Sets the ID number of the HA cluster. Defaults to 1
```
Comment (samccann) [inline lib/ansible/modules/network/fortimanager/fmgr_ha.py]:
```suggestion
      - Sets the ID number of the HA cluster. Defaults to 1.
```

File: lib/ansible/modules/network/fortimanager/fmgr_ha.py [hunk 7]
```diff
@@ -0,0 +1,404 @@
...
+    required: false
+    default: 1
+  fmgr_ha_hb_threshold:
+    description:
+      - Sets heartbeat lost threshold (1-255)
```
Comment (samccann) [inline lib/ansible/modules/network/fortimanager/fmgr_ha.py]:
```suggestion
      - Sets heartbeat lost threshold (1-255).
```

File: lib/ansible/modules/network/fortimanager/fmgr_ha.py [hunk 8]
```diff
@@ -0,0 +1,404 @@
...
+    required: false
+    default: 3
+  fmgr_ha_hb_interval:
+    description:
+      - Sets the heartbeat interval (1-255)
```
Comment (samccann) [inline lib/ansible/modules/network/fortimanager/fmgr_ha.py]:
```suggestion
      - Sets the heartbeat interval (1-255).
```

File: lib/ansible/modules/network/fortimanager/fmgr_ha.py [hunk 9]
```diff
@@ -0,0 +1,404 @@
...
+    required: false
+    default: 5
+  fmgr_ha_file_quota:
+    description:
+      - Sets the File quota in MB (2048-20480)
```
Comment (samccann) [inline lib/ansible/modules/network/fortimanager/fmgr_ha.py]:
```suggestion
      - Sets the File quota in MB (2048-20480).
```

File: lib/ansible/modules/network/fortimanager/fmgr_ha.py [hunk 10]
```diff
@@ -0,0 +1,404 @@
...
+    password: "{{ password }}"
+    fmgr_ha_mode: "standalone"
+
+- name: ADD FORTIMANAGER HA PEER
+  fmgr_ha:
+    host: "{{ inventory_hostname }}"
+    username: "{{ username }}"
+    password: "{{ password }}"
+    fmgr_ha_peer_ipv4: "10.7.220.36"
+    fmgr_ha_peer_sn: "FMG-VM0A17002010"
```
Comment (samccann) [inline lib/ansible/modules/network/fortimanager/fmgr_ha.py]: I don't know if this is a real serial number or if you want to make this a dummy number such as FMG-VM0A123456789

Comment (ftntcorecse) [reply lib/ansible/modules/network/fortimanager/fmgr_ha.py]: Good point. Changed it.

**Comment by ftntcorecse** [comment]
@samccann Sorted out edits and minor issues. Should be rdy.

**Review by samccann — APPROVED** [review]
docs part looks good.

**Review by NilashishC — APPROVED** [review]

**Comment by NilashishC** [comment]
LGTM

**Pull request merged**